### PR TITLE
Feature/mds 4388 edit nod

### PR DESCRIPTION
**Pull request by matbusby-fw**
## Objective 

[MDS-4388](https://bcmines.atlassian.net/browse/MDS-4388)

- Added an Edit modal for NODS
- updated redux action creators with update endpoint

## Additional Information / Context 

image

### CODE REVIEW

File: services/core-api/app/api/mines/notice_of_departure/resources/notice_of_departure.py [hunk 1]
```diff
@@ -40,11 +40,18 @@ def patch(self, mine_guid, nod_guid):
...
+            help='Notice of Departure type',
+            location='json',
+            required=True,
+            store_missing=False)
         data = parser.parse_args()
 
         nod.nod_title = data.get('nod_title')
         nod.nod_description = data.get('nod_description')
-        nod.nod_type = NodType.potentially_substantial # todo: removed when this feature if being built
+        nod.nod_type = NodType[data.get('nod_type')]
```
Comment (hitankar-freshworks) [inline services/core-api/app/api/mines/notice_of_departure/resources/notice_of_departure.py:54]: maybe have default if the nod-type is not part of the payload.
`nod.nod_type = NodType[data.get('nod_type', NodType['pending_review'])]`

Comment (matbusby-fw) [reply services/core-api/app/api/mines/notice_of_departure/resources/notice_of_departure.py:54]: :+1: updated to `nod.nod_type = NodType[data.get('nod_type', 'pending_review')]`

File: services/core-api/app/api/mines/notice_of_departure/resources/notice_of_departure.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def patch(self, mine_guid, nod_guid):
 
         nod.nod_title = data.get('nod_title')
         nod.nod_description = data.get('nod_description')
-        nod.nod_type = NodType[data.get('nod_type', 'pending_review')]
+        nod.nod_type = NodType[data.get('nod_type')]
         nod.nod_status = NodStatus.pending_review
```
Comment (hitankar-freshworks) [inline services/core-api/app/api/mines/notice_of_departure/resources/notice_of_departure.py:55]: This is going to set nod to pending_review always.

**Review by hitankar-freshworks — APPROVED** [review]
One comment.. looks good otherwise

File: services/core-api/app/api/mines/notice_of_departure/resources/notice_of_departure.py (hunk 2 above)
Comment (matbusby-fw) [reply services/core-api/app/api/mines/notice_of_departure/resources/notice_of_departure.py:55]: Yes, that's how it had already been set.  Will be addressing that in the upcoming PR.

**Pull request merged**